app/src/main/java/com/github/polybooks/core/database/implementation/OLBookDatabase.kt

```diff
@@ -80,18 +81,9 @@ class OLBookDatabase(private val url2json : (String) -> CompletableFuture<JsonEl
         override fun getAll(): CompletableFuture<List<Book>> {
             if (empty) return CompletableFuture.completedFuture(Collections.emptyList())
             else {
-                assert(isbn != null)
-                val url = isbn2URL(isbn!!)
-                return url2json(url)
-                        .thenApply { parseBook(it) }
-                        .thenCompose { updateBookWithAuthorName(it) }
-                        .thenApply { listOf(it) }
-                        .exceptionally { exception ->
-                            if (exception is CompletionException && exception.cause is FileNotFoundException)
-                                return@exceptionally Collections.emptyList<Book>()
-                            else if (exception is CompletionException) throw exception.cause!!
-                            else throw exception
-                        }
+                assert(isbns != null)
```

Comment:
here a warning says "Assertions are never enabled in Android" so maybe remove this line